source/class/qx/tool/migration/M6_0_0.js

```diff
@@ -0,0 +1,183 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2021 The authors
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * Christian Boulanger (info@bibliograph.org, @cboulanger)
+
+************************************************************************ */
+
+const process = require("process");
+const path = require("upath");
+const semver = require("semver");
+const fs = qx.tool.utils.Promisify.fs;
+
+/**
+ * Migration class for updating from v5 to v6
+ */
+qx.Class.define("qx.tool.migration.M6_0_0", {
+  extend: qx.core.Object,
+  implement: qx.tool.migration.IMigration,
+  members: {
+    /**
+     * @inheritDoc
+     * @see {qx.tool.migration.IMigration#migrate()}
+     */
+    migrate: async function (options = {}) {
+      let {announceOnly} = options;
+      let needFix = false;
+      let pkg = qx.tool.cli.commands.Package;
+      let cwd = process.cwd();
+      let migrateFiles = [[path.join(cwd, pkg.lockfile.filename), path.join(cwd, pkg.lockfile.legacy_filename)], [path.join(cwd, pkg.cache_dir), path.join(cwd, pkg.legacy_cache_dir)], [path.join(qx.tool.cli.ConfigDb.getDirectory(), pkg.package_cache_name), path.join(qx.tool.cli.ConfigDb.getDirectory(), pkg.legacy_package_cache_name)]];
+      if (qx.tool.migration.Utils.checkFilesToRename(migrateFiles).length) {
```

Comment:
Some comments in here saying what this code is trying to accomplish would be really useful. It's not obvious.

Reply:
The migration code has been modularized, with each step in one separate method- the name should tell you what it does: https://github.com/cboulanger/qooxdoo/blob/refactor_migration/source/class/qx/tool/migration/M6_0_0.js